### DISCUSSION TITLE
Profile packs versions

### DISCUSSION THREAD

**Discussion by SGe-GH** [krosmaker/krosmaker#51] [Ideas]
Since we are using github, it might be appropriate to use its ability to keep track of versions for the profiles published here.

That way if one of them happens to be unbalanced or in need of tuning, it will be easy to manage its changes (the changes in themselves and the explanations of the changes made to a profile).

When I write this, it's really the profiles when they are released and made available for the community, since it somehow makes them official (fanmade-wise).

**Comment by czyzby** [comment]
I've thought about keeping the JSON files in a repository, but they're pretty big since they store the figurine and artwork images. Some text editors might have trouble opening them at all. That said, I doubt we'd hit any GitHub storage limits, but the card repository might be pretty slow.

As for the version itself, do you think it should be displayed somewhere on the card or purely for editors' information? I could add another field into the `Krosmaster` or `Storage` form and display it at the back of the card next to the watermark.

**Comment by SGe-GH** [comment]
I was more thinking about versioning the final result like in your post https://github.com/krosmaker/krosmaker/discussions/46 not the json files themselves (I don't know which one is better for storage on github). More like "Here is the official release of Yugo; this release is v1.0" and later if it needs balancing, then it can be released in version 1.1 with its release note stating that "5 MP is a little too much in addition to the spell enabling him to jump next to his target". More like a repository for the tool and a repository for the profiles collection.

Concerning adding a field in the tool itself, I add this idea also at first, but I abandoned it because adding a version number next to the actual watermark might be a source of confusion. Let's say, you made Missiz Burnz profile and put a version number on its back (e.g. v1.2). Months later someone else makes changes in the stats and shares this new profile elsewhere (on facebook groups for example) but with the same back. And now people will share and use two different profiles with the same back displaying the same version number v1.2.

If you add a version number as watermarck on images, it would need to appear on the front image also I think. Also it would be useful to not just have something like "v1.0" but more like "MC v1.0" for example (a Set name/letters followed by the version).

The idea is not to "copyright" it but to keep track of which profile we put our hands on (which "set/collection/source/author" does it come from and then in this set, is this the most up to date version).

**Reply by czyzby** [reply]
> I abandoned it because adding a version number next to the actual watermark might be a source of confusion.

I think it's impossible to avoid this, even if we don't include the version on the card. As long as we publish the tool and the editable files, anyone can post their own version of each profile online. Even if you compare the cards that I've published on Twitter, BGG and here you might find some slight differences due to balancing changes, tool updates and so on.

Besides, a built-in versioning tool might help other unofficial fan projects. I wouldn't dismiss that idea outright.

We should find the right place to display it though. Maybe the bottom right corner at the back of the card on the golden frame? I don't want to modify the card fronts, because no watermarks or annotations are displayed on the original cards either. Actually, the official cards seem to have some sort of card ID/version on the right side of the artwork, where Krosmaker currently displays a subtle `@Krosmaker` watermark. I guess we could replace out watermark - but I'd like to avoid that to make it clear that the profiles are not official - or add the version next to it, but that might be confusing as well, as the users might think it's the tool version, not the card's

> I was more thinking about versioning the final result like in your post #46 not the json files themselves (I don't know which one is better for storage on github).

We could do both - store the exported images and the editable JSON files in a repository.

**Reply by czyzby** [reply]
Just so you know, I've added the card version to the main Krosmaster view. The version is displayed at the bottom right corner of the artwork frame on the back of the card. I'm also in the process of creating a separate repository for the cards, so we can use Git to keep track of their revisions as well.
